assets/app/scripts/services/userstore.js

```diff
@@ -6,69 +6,141 @@ angular.module('openshiftConsole')
+.provider('SessionStorageUserStore', function() {
   this.$get = function(){
     var debug = false;
-    var userkey = "user";
-    var tokenkey = "token";
+    var userkey = "SessionStorageUserStore.user";
+    var tokenkey = "SessionStorageUserStore.token";
     return {
+      available: function() {
+        try {
+          var x = new Date().getTime();
+          sessionStorage['test'] = x;
+          var y = sessionStorage['test'];
+          sessionStorage.removeItem('test');
+          return x == y;
+        } catch(e) {
+          return false;
+        }
+      },
       getUser: function(){
         try {
           var user = JSON.parse(sessionStorage[userkey]);
-          if (debug) { console.log("SessionUserStore.getUser", user); }
+          if (debug) { console.log("SessionStorageUserStore.getUser", user); }
           return user;
         } catch(e) {
-          if (debug) { console.log("SessionUserStore.getUser", e); }
+          if (debug) { console.log("SessionStorageUserStore.getUser", e); }
           return null;
         }
       },
       setUser: function(user) {
         if (user) {
-          if (debug) { console.log("SessionUserStore.setUser", user); }
+          if (debug) { console.log("SessionStorageUserStore.setUser", user); }
           sessionStorage[userkey] = JSON.stringify(user);
         } else {
-          if (debug) { console.log("SessionUserStore.setUser", user, "deleting"); }
+          if (debug) { console.log("SessionStorageUserStore.setUser", user, "deleting"); }
           sessionStorage.removeItem(userkey);
         }
       },
       getToken: function() {
         try {
           var token = sessionStorage[tokenkey];
-          if (debug) { console.log("SessionUserStore.getToken", token); }
+          if (debug) { console.log("SessionStorageUserStore.getToken", token); }
           return token;
         } catch(e) {
-          if (debug) { console.log("SessionUserStore.getToken", e); }
+          if (debug) { console.log("SessionStorageUserStore.getToken", e); }
           return null;
         }
       },
       setToken: function(token) {
         if (token) {
-          if (debug) { console.log("SessionUserStore.setToken", token); }
+          if (debug) { console.log("SessionStorageUserStore.setToken", token); }
           sessionStorage[tokenkey] = token;
         } else {
-          if (debug) { console.log("SessionUserStore.setToken", token, "deleting"); }
+          if (debug) { console.log("SessionStorageUserStore.setToken", token, "deleting"); }
           sessionStorage.removeItem(tokenkey);
         }
       }
     }
   };
+})
+.provider('LocalStorageUserStore', function() {
+  this.$get = function(){
+    var debug = false;
+    var userkey = "LocalStorageUserStore.user";
+    var tokenkey = "LocalStorageUserStore.token";
+    return {
+      available: function() {
+        try {
+          var x = new Date().getTime();
+          localStorage['test'] = x;
```

Comment:
since localStorage crosses tabs should we be testing on a key that is namespaced to openshift in case we have to play nice with other products on the same domain but different root paths?